src/stores/store.ts

```diff
@@ -294,3 +311,142 @@ export const infoStore = defineStore('info', {
     }
   }
 });
+
+export const itemsStore = defineStore('items', {
+  state: () => ({
+    selectedItem: {} as item | null,
+    shops: [{ name: 'Default', items: [] }] as shop_list[],
+    activeShop: 0,
+    generating: false
+  }),
+  getters: {
+    getSelectedItem: (state) => state.selectedItem,
+    getShops: (state) => state.shops,
+    getActiveShop: (state) => state.shops[state.activeShop],
+    getGenerating: (state) => state.generating,
+    getTotalCost: (state) => {
+      let cost = 0;
+      state.shops[state.activeShop].items.forEach((item) => {
+        for (let i = 0; i < item.quantity; i++) {
+          cost += item.price;
+        }
+      });
+      return cost;
+    }
+  },
+  actions: {
+    setSelectedItem(newSelectedItem: item) {
+      this.selectedItem = newSelectedItem;
+    },
+    removeSelectedItem() {
+      this.selectedItem = null;
+    },
+    setShopList(newShopList: shop_list[]) {
+      this.shops = newShopList;
+    },
+    clearShop() {
+      this.shops[this.activeShop].items.splice(0, this.shops[this.activeShop].items.length);
+    },
+    clearItem(item: min_item) {
+      const index = this.shops[this.activeShop].items.indexOf(item);
+      this.shops[this.activeShop].items.splice(index, 1);
+    },
+    addToShop(item: min_item, index?: number) {
+      if (index! >= 0) {
+        item.quantity++;
+        this.shops[this.activeShop].items.splice(index!, 1, item);
+      } else {
+        this.shops[this.activeShop].items.push(item);
+      }
+    },
+    removeFromShop(index: number) {
+      if (this.shops[this.activeShop].items[index].quantity > 1) {
+        this.shops[this.activeShop].items[index].quantity--;
+      } else {
+        this.shops[this.activeShop].items.splice(index, 1);
+      }
+    },
+    changeActiveShop(shopIndex: number) {
+      if (shopIndex >= this.shops.length || shopIndex < 0) {
+        this.activeShop = 0;
+      } else {
+        this.activeShop = shopIndex;
+      }
+    },
+    addShop(shopName: string) {
+      this.shops.push({ name: shopName, items: [] });
+      this.activeShop = this.shops.length - 1;
+    },
+    removeShop() {
+      this.shops.splice(this.activeShop, 1);
+      this.activeShop = 0;
+      if (this.shops.length <= 0) {
+        this.shops = [{ name: 'Default', items: [] }];
+      }
+    },
+    getShopIndex(shopName: string): number {
+      return this.shops.map((shop) => shop.name).indexOf(shopName);
+    },
+    updateShop(shopName: string, newItems: min_item[]) {
+      const shopIndex = this.getShopIndex(shopName);
+      if (shopIndex >= 0) {
+        this.shops[shopIndex].items = newItems;
+      }
+    },
+    updateShops(newShops: shop_list[]) {
+      this.shops = newShops;
+    },
+    setGenerating(newGenerating: boolean) {
+      this.generating = newGenerating;
+    },
+    getFormattedPrice(price: number) {
+      if (price < 10) {
+        return price + ' cp';
+      } else if (price >= 10 && price < 100) {
```

Comment:
You could keep this for the sake of clarity, but you are already sure that price is equal or higher than 10. (if else clause)